test/src/common/harnessTls.c

```diff
@@ -104,23 +105,13 @@ harnessTlsServerInitDefault(void)
 
 /***********************************************************************************************************************************
 Expect an exact string from the client
-
-This is a very unforgiving function and short input will leave the server hanging.  Definitely room for improvement here.
 ***********************************************************************************************************************************/
 void
 harnessTlsServerExpect(const char *expected)
 {
     Buffer *buffer = bufNew(strlen(expected));
-    int readBytes = 0;
 
-    // Read expected bytes
-    do
-    {
-        int lastBytes = SSL_read(testClientSSL, bufRemainsPtr(buffer), (int)bufRemains(buffer));
-        readBytes += lastBytes;
-        bufUsedSet(buffer, (size_t)readBytes);
-    }
-    while (bufRemains(buffer));
+    ioRead(tlsSessionIoRead(testServerSession), buffer);
 
     // Treat and ? characters as wildcards so variable elements (e.g. auth hashes) can be ignored
```

Comment:
I know this is an old comment, but Treat "what" and ? characters?

Reply:
Should be "treat any ? characters". I'll fix that.

Reply:
Type committed at 02ebba02b2 and the rest at 9ffa2c618fe54.